### DISCUSSION TITLE
Unable to create custom 404 page

### DISCUSSION THREAD

**Discussion by imheinrichlunge** [vercel/vercel#9567] [Help]
Hi everyone, I have a very simple website with just HTML and CSS and I'm trying to update the legacy "routes" parameter to create a custom 404 page, but I am unable to do it.

I'm trying to update it based on this [article](https://vercel.com/guides/custom-404-page#what-if-i-need-to-name-my-404-file-something-different) and following the recommendations of this [article](https://vercel.com/docs/concepts/projects/project-configuration#upgrading-legacy-routes) and this [one](https://vercel.com/docs/concepts/projects/project-configuration) using `redirects` and `rewrites`.

My json looks like this (I tried removing the `has` property, but it doesn't work either:
``` json
{
    "cleanUrls": true,
    "rewrites": [
        {
            "source": "/(.*)",
            "has": [
                {
                    "type": "header",
                    "key": "x-vercel-error",
                    "value": "NOT_FOUND"
                }
            ],
            "destination": "/404.html"
        }
    ]
}
```

Thank you in advance for helping me 😉

**Comment by ivoilic** [comment]
Hey there! Can you clarify why aren't you just using exactly what's in the first example you linked instead of rewrites?

{
  "routes": [
    { "handle": "filesystem" },
    { "src": "/(.*)", "status": 404, "dest": "/other-404.html" }
  ]
}

**Reply by imheinrichlunge** [reply]
Because I can't use legacy parameters with new ones in the same file, and `cleanUrls` is just perfect for my workflow, the only one would be not to use `cleanUrls`.
